client/state/oauth2-clients/ui/selectors.js

```diff
@@ -17,7 +19,12 @@ export function getCurrentOAuth2ClientId( state ) {
  * @returns {Object}          OAuth2 client data
  */
 export const getCurrentOAuth2Client = ( state ) => {
-	const currentClientId = getCurrentOAuth2ClientId( state );
+	let currentClientId = getCurrentOAuth2ClientId( state );
+
+	// Enables Gravatar related OAuth2 clients to use the Gravatar login flow.
+	if ( isGravatarFlowOAuth2Client() ) {
+		currentClientId = GRAVATAR_CLIENT_ID;
```

Comment:
I don't think we should do this; it's almost impersonating the Gravatar app. In the future we might want to show the current app logo / title / information. 

I would instead update the checks we make to return true for gravatar-created apps OR return true if the parameter `gravatar_flow` exists.

Reply:
Since we have the `oauth2Client.source`. So, we have used the optimal solution instead.